extensions/rename.md

```diff
@@ -0,0 +1,114 @@
+---
+title: IRCv3 `rename` Extension
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "Peter Powell"
+    email: "petpow@saberuk.com"
+    period: "2017"
+---
+
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+Software implementing this work-in-progress specification MUST NOT use the
+unprefixed `rename` capability name. Instead, implementations SHOULD use the
+`draft/rename` capability name to be interoperable with other software
+implementing a compatible work-in-progress version.
+
+The final version of the specification will use an unprefixed capability name.
+
+## Introduction
+
+Currently there is no reasonable way to rename channels. All existing methods
+involve closing down the channel and redirecting it to the a new one. This
+is undesirable as access and ban lists needs to be manually copied. It also
+causes problems for client logging as history is split across multiple files.
+
+This specification introduces a method of natively renaming channels that
+does not suffer from these faults.
+
+## Architecture
+
+### Capabilities
+
+This specification introduces the `draft/rename` capability. This capability
+informs the server that the client is capable of handling `RENAME` messages.
+
+### Messages
+
+This specification adds the `RENAME` C2S and S2C message. This message informs
+a client that the name of a channel has been changed.
+
+The `RENAME` message has between two and three parameters. The first parameter
+is the current channel name, the second parameter is the new channel name, and
+the optional third parameter is a reason for renaming the channel.
+
+Server implementations SHOULD NOT allow channels to be converted between types
```

Comment:
What is the reason behind preventing changing these types?

I apologise if this is mentioned somewhere, would (freenode) `##` and `#` could as different types?

Reply:
Nah, those would be the same type. Converting between types would be like converting a normal channel (`#`) to a local channel (`&`), and vice-versa. Different channel types tend to act in completely different ways, so I can understand this restriction.

Reply:
Would `#&` and `##` be the same type? I though this could depend on server implementation details.

Reply:
Yes. The significant character is the first character. `&whatever` for some implementations is a perfectly valid channel name.

Reply:
Actually, as an example - znc has this channel concept called "partyline", where channels can be prepended with `~` and it is a different "type" of channel than the one for the IRC network you're connected to. In this case, it definitely wouldn't work because users in the "partyline" channels might not actually be able to join the channels on the server your znc session is connected to.

Reply:
Yes and no... Servers can of course be written to do all sorts of crazy things (like making channels global depending on a mode) – the sky is the limit. But as far as the _documented_ IRC protocol goes, only the first character is recognized as type distinguisher, and IMHO it's perfectly valid for a spec to assume that.